xlators/cluster/afr/src/afr-self-heal-common.c

```diff
@@ -2391,7 +2391,7 @@ afr_selfheal_unlocked_inspect(call_frame_t *frame, xlator_t *this, uuid_t gfid,
                 *metadata_selfheal = _gf_true;
         }
 
-        if (IA_ISREG(first.ia_type) &&
+        if (IA_ISREG(first.ia_type) && !(AFR_IS_ARBITER_BRICK(priv, i)) &&
```

Comment:
Will the stale entry inside indices be purged with this change when you run heal info or by the shd during heal?

Reply:
Currently this is not happening with just this change. Only after the brick gets restarted, these stale entries are getting purged. I will check why it is not happening, and try to fix it, which will address the previous comment as well.

Reply:
I could not give much time to investigate this. But what I found sometime back was, during the index crawl entry purge was happening only when the ret value is set to ENOENT or ESTALE. But in this case the ret value will be 2 indicating it does not need any heal, and calls afr_shd_zero_xattrop(), which should ideally remove the entry from the index. I will try to work on it this week.